api/cloudsync_api.yaml

```diff
@@ -0,0 +1,139 @@
+              $ref: '#/components/schemas/subscribebody'
+
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubscribeResponse'
+  /api/v1/orchestration/cloudsyncmgr/publish:
+    post:
+      tags:
+        - CloudSync
+      summary: ""
+      description: ""
+      operationId: publish
+      requestBody:
+        description: publish to MQTT broker running on url for the given topic
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/publishbody'
+
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PublishResponse'
+  /api/v1/orchestration/cloudsyncmgr/getsubscribedata/{host}/{topic}/{appID}:
+   get:
+      tags:
+        - CloudSync
+      summary: ""
+      description: ""
+      operationId: getSubscribeddata
+      parameters:
+        - name: host
+          in: path
+          description: url of the mqtt broker
+          required: true
+          schema:
+            type: string
+        - name: topic
+          in: path
+          description: topic for which the message is required
+          required: true
+          schema:
+            type: string
+        - name: appID
+          in: path
+          description: id of the app which has subscribed to mentioned topic
+          required: true
+          schema:
+            type: string
+           
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetDataResponse'      
+        '404':
+          description: Page not found
+    
+components:
+  schemas:
+    subscribebody:
+      properties:
+        appid:
+          type: string
+          example: com.samsung.facerecog
+        topic:
+          type: string
+          example: doggie
+        url:
+          type: string
+          example: ec2-184-73-83-74.compute-1.amazonaws.com
+    publishbody:
+      properties:
+        appid:
+          type: string
+          example: com.samsung.facerecog
+        payload:
+          type: string
+          example: Test Data from Sample app
+        topic:
+          type: string
+          example: doggie
+        url:
+          type: string
+          example: ec2-184-73-83-74.compute-1.amazonaws.com
+    SubscribeResponse:
+      properties:
+        Message:
+          type: string
+          example: Successfully subscribed to the topic homeedge/livingroom 
+    PublishResponse:
+      properties:
+        Message:
+          type: string
+          example: Data published successfully to Cloudtcp://broker.hivemq.com:1883
```

Comment:
Please use `publishbody url` instead of hivemq address.

Reply:
done!!